src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlClientEventSource.NetCoreApp.cs

```diff
@@ -0,0 +1,384 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics.Tracing;
+using System.Threading;
+
+namespace Microsoft.Data.SqlClient
+{
+    internal partial class SqlClientEventSource : EventSource
```

Comment:
Hi Shay,
Among the event counter types, just `EventCounter` supports `.NetCore 2.0` and above. This is the only reason for choosing it for now.

Reply:
I see... I remember having a similar discussion back when first implementing this in Npgsql. As this is a new feature and the new polling counters are already available in .NET Core 3.0, I'd advise going with the latter... Otherwise you're penalizing perf of modern applications using perf counters for the sake of having the new perf counters in pre-3.0 .NET...

Another thought is that if you have actual demand for this in pre-.NET 3.0, it's always possible to use conditional complication, using the polling counters in >3.0 and the current code when targeting older TFMs (though I'd really wait to see that this is required).

Reply:
Hi Shay,

We support NET Core 2.1 applications and aim to provide full feature support in all supported environments if possible. However, it's not clearly defined whether it is "required" for .NET Core 2.1 or not. It does keep us little behind from using latest features, but that's only for best feature compatibility and user experience.

But if there're new features that brings considerable performance difference/user experience, we're definitely in favor of splitting implementation and keeping it TFM specific, like we've been doing always. Our only ask would be does it make that big of a difference that we maintain double code for .NET Core 2.1 and 3.1+? Or we can continue it as it is for code simplicity, and when .NET Core 2.1 goes out of support, we can update this to new Counter classes for future.

Reply:
On the contrary,
I see value in splitting classes as it is in this case as there's a lot to offer in .NET Core 3.1 and .NET Standard 2.1 DLLs.

cc @DavoudEshtehari

Reply:
@cheenamalhotra I understand, but the .NET Core performance counters feature itself is new - the dotnet counters tool wasn't even introduced until .NET Core 3.0, if memory serves. The feature called "performance counters" in .NET Framework is similar in goals, but entirely different in the way it works and incompatible with .NET Core, so perf counters in .NET Core can be considered essentially a new feature in .NET Core 3.0, IMHO.

In other words, I'd be quite surprised if anyone using versions of .NET Core lower than 3.0 would be interested in this, and it seems like a pity to (significantly) compromise perf for users of .NET Core 3.0 and above (or IMHO even to maintain two implementations).

Reply:
Thanks for context @roji 👍 
Linking documentation for readers/reviewers: [Event Counters](https://docs.microsoft.com/en-us/dotnet/core/diagnostics/event-counters)

Even though the APIs exist in .NET Core 2.1, from official Microsoft documentation it sounds like we can consider this for .NET Core 3.1 and .NET Standard 2.1 onwards.

@David-Engel your thoughts?

Reply:
Resolving as implementation targeted for netcoreapp3.1+ and netstandard2.1+